apps/src/code-studio/components/progress/ScriptOverviewHeader.jsx

```diff
@@ -17,15 +21,44 @@ class ScriptOverviewHeader extends Component {
     plcHeaderProps: PropTypes.shape({
       unitName: PropTypes.string.isRequired,
       courseViewPath: PropTypes.string.isRequired,
-    })
+    }),
+    announcements: PropTypes.arrayOf(
+      PropTypes.shape({
+        notice: PropTypes.string.isRequired,
+        details: PropTypes.string.isRequired,
+        link: PropTypes.string.isRequired,
+        type: PropTypes.oneOf(Object.values(NotificationType)).isRequired,
+      })
+    ),
+    viewAs: PropTypes.oneOf(Object.values(ViewType)).isRequired,
+    isSignedIn: PropTypes.bool.isRequired,
+    isVerifiedTeacher: PropTypes.bool.isRequired,
+    hasVerifiedResources: PropTypes.bool.isRequired,
   };
 
   componentDidMount() {
     $('#lesson').appendTo(ReactDOM.findDOMNode(this.protected));
   }
 
   render() {
-    const { plcHeaderProps } = this.props;
+    const {
+      plcHeaderProps,
+      announcements,
+      viewAs,
+      isSignedIn,
+      isVerifiedTeacher,
+      hasVerifiedResources,
+    } = this.props;
+
+    let verifiedResourcesAnnounce = [];
+    if (!isVerifiedTeacher && hasVerifiedResources) {
```

Comment:
Just a thought: This seems like logic that could live as a helper/selector in a redux module, rather than in this component, so the view just gets the list of notifications and displays it.

Reply:
What I'd really like to happen is that as soon as we know that you're an unverified teacher and the script has verified resources, we do `store.dispatch(addAnnouncement(verifiedResourcesAnnounce))`. The reason that's not easy is that there are multiple places where we get signin info async, and getting this check right in all those places could be difficult.

The one challenge with moving this into a selector is that we're dealing with two difference reducers (state.verifiedTeachers and state.scriptAnnouncements). The selector would then either need to live in some shared location, or be passed in one of the two sets of state. Both are doable, and if this were used in more than one place, I think moving this would make sense, but as it is, I think this is okay.